### PR TITLE
[FIX] libc++: performance tests

### PR DESCRIPTION
**Pull request by eseiler**

### CODE REVIEW

File: test/performance/search/views/view_minimiser_hash_benchmark.cpp [hunk 1]
```diff
@@ -98,7 +98,7 @@ void compute_minimisers(benchmark::State & state)
             // Use random seed to randomise order on forward strand.
             auto forward = seq | seqan3::views::kmer_hash(shape)
                          | std::views::transform(
-                               [seed](uint64_t const i)
+                               [](uint64_t const i)
                                {
                                    return i ^ seed;
                                });
```
Comment (smehringer) [inline test/performance/search/views/view_minimiser_hash_benchmark.cpp:104]: how is `seed` available now?

File: test/performance/utility/simd/views/iota_simd_benchmark.cpp [hunk 2]
```diff
@@ -88,7 +88,7 @@ struct for_loop_with_simd_add_benchmark
     void operator()(index_simd_t & count) const
     {
         index_simd_t simd_index{};
-        for (size_t index = 0; index < end_index; ++index, ++simd_index)
+        for (size_t index = 0; index < end_index; ++index, simd_index += 1)
```
Comment (smehringer) [inline test/performance/utility/simd/views/iota_simd_benchmark.cpp:91]: `++` did not work? seems so basic

Comment (eseiler) [reply test/performance/utility/simd/views/iota_simd_benchmark.cpp:91]: Libc++ says it implements `++` for Simd types, but it doesn't work :D

File: test/performance/search/views/view_minimiser_hash_benchmark.cpp (hunk 1 above)
Comment (eseiler) [reply test/performance/search/views/view_minimiser_hash_benchmark.cpp:104]: It's implicitly captured because it's const.

**Review by smehringer — APPROVED** [review]

**Pull request merged**